docs/concepts.rst

```diff
@@ -228,6 +228,84 @@ Not like this, where the join task is skipped
 
 .. image:: img/branch_bad.png
 
+SubDAGs
+'''''''
+
+Sometimes sections of your workflow can be grouped into logical units that
+can also be represented as DAGs. The ``SubDagOperator`` is useful in these
+situations.
+
+For example, consider the following DAG:
+
+.. image:: img/subdag_before.png
+
+We can combine all of the parallel ``task-*`` operators into a single SubDAG,
+so that the resulting DAG resembles the following:
+
+.. image:: img/subdag_after.png
+
+Note that SubDAG operators should contain a factory method that returns a DAG
+object. This will prevent the SubDAG from being treated like a separate DAG in
+the main UI. For example:
+
+.. code:: python
+
+  #dags/subdag.py
+  from airflow.models import DAG
+  from airflow.operators import DummyOperator
+
+
+  # Dag is returned by a factory method
+  def sub_dag(start_date, schedule_interval):
+    dag = DAG(
+      'main_dag.dummy_operator',
+      schedule_interval=schedule_interval,
+      start_date=start_date,
+    )
+
+    dummy_operator = DummyOperator(
+      task_id='dummy_task',
+      dag=dag,
+    )
+
+    return dag
+
+This SubDAG can then be referenced in your main DAG file:
+
+.. code:: python
+
+  # main_dag.py
+  from datetime import datetime, timedelta
+  from airflow.models import DAG
+  from airflow.operators import SubDagOperator
+  from dags.subdag import sub_dag
+
+
+  main_dag = DAG(
+    dag_id='main_dag',
+    schedule_interval=timedelta(hours=1),
+    start_date=datetime(2016, 1, 1)
+  )
+
+  sub_dag = SubDagOperator(
+    subdag=sub_dag(main_dag.start_date, main_dag.schedule_interval),
+    task_id='dummy_task',
+    dag=main_dag,
+  )
+
+Some other tips when using SubDags:
+
+-  by convention, a SubDAG's ``dag_id`` should be prefixed by its parent and
+   a dot. As in ``parent.child``.
+-  share arguments between the main DAG and the SubDAG by passing arguments to
+   the SubDAG operator (as demonstrated above).
+-  SubDAGs must have a schedule and be enabled. If the SubDAG's schedule is
+   set to ``None`` or ``@once``, the SubDAG will succeed without having done
+   anything.
+-  you can click into a SubDagOperator from the graph view of the main DAG to
+   show the tasks contained within the SubDAG.
+
```

Comment:
- refrain from using `depends_on_past=True` in tasks within the subdag as this can be confusing
- it's possible to specify an `executor` for the subdag, it's common to use SequentialExecutor if you want to run the subdag in-process and effectively limit its parallelism to 1. Using LocalExecutor can be bad as it may over subscribe your worker, running multiple tasks in a single slot
- subdags are perfect for repeating patterns, and defining a function that returns a DAG object is a nice design pattern. For example, Airbnb uses the stage-check-echange pattern when loading data where the data is staged in a temporary table, data quality checks are performed against that table and once the checks all pass the partition is exchanged into production
- clearing a `SubDagOperator` also clears the state of the tasks within
- marking success on a `SubDagOperator` does not affect the state of the tasks within